### ISSUE TITLE
Understand FeatureFinderIdentification param `-detect:mapping_tolerance`

### ISSUE THREAD

**Issue by zhuchcn** [OpenMS/OpenMS#6547]
I'm trying to understand the `detect:mapping_tolerance` parameter of FeatureFinderIdentification. According to the documentation, this parameter is the allowed difference of RT between the PSM and feature. And seems like it's taking the distance of the PSM to the upper or lower bond of the feature box? My question is then, sure this parameter be ever set? Is it OK to assign a peptide to a feature when the RT of the PSM is outside of the feature box?

**Comment by jpfeuffer** [comment]
Hi!

No, the tolerance is relative to the apex of the feature. Therefore it makes sense.
Otherwise the description of its possible values: "Absolute value in seconds if 1 or greater, else relative to the RT span of the feature." would not make sense.

**Comment by zhuchcn** [comment]
Got it! That makes a lot of sense. We did find that the number of peptides quantified by ffid is maximized when `mapping_tolerance` is set to be the same as `peak_width`. Should we set these two parameters the same? Or should we set it to 0.5 so it will be always same as `peak_width` (if I understand it correctly)? The default value of this param is 0.0, which doesn't make much sense. Should it be changed to an appropriate value such as 0.5?

**Comment by zhuchcn** [comment]
I'm reading the source code of `FeatureFinderIdentificationAlgorithm::annotateFeatures_` here, and it seems like it is getting the peptide identifications between the lower and upper bond + mapping tolerance. So it is the allowed RT difference from the upper/lower RT of the feature?

https://github.com/OpenMS/OpenMS/blob/3d5c3462ca77fbcbb2b77ca9f57909a002625c01/src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp#L1157-L1179

**Comment by jpfeuffer** [comment]
So, the problem is that FFID has a mode for transferring IDs across samples.
It may be that the parameter is only used in this case, where a wider tolerance (starting from rt_min and rt_max) makes sense.

**Comment by jpfeuffer** [comment]
That is a good question. I tagged the creator of the tool.
Have you ever looked at examples in TOPPView? Can you show a picture of such an example, were a (most likely correct) MS2 spectrum is not mapped to a feature because the tolerance was too low.

**Comment by hendrikweisser** [comment]
Creator of the tool here.

> And seems like it's taking the distance of the PSM to the upper or lower bond of the feature box?

This is correct.

> No, the tolerance is relative to the apex of the feature.

This is *not* correct.

The parameter exists for the following reason: The exact boundaries (RT start/end) of a feature are hard to detect, so the underlying library (OpenSWATH) may sometimes generate features that are slightly "too short" (based on my experience, it errs towards "too short" rather than "too long"). However, depending on your instrument settings, MS2 spectra - and thus PSMs - may be triggered very early in the elution (and, depending on exclusion time, potentially not again for that ion/feature). These PSMs may fall (slightly) outside the feature boundary, so the point of the parameter is to add some tolerance for the mapping. Note that the default is zero, so it's really intended to be used (with a positive value) only when inspection of the data (e.g. in TOPPView) suggests that it's needed.

**Comment by hendrikweisser** [comment]
I should add that features being slightly too short typically doesn't matter for quantification, because the signal intensity at the tails is very low.

**Comment by zhuchcn** [comment]
Really appreciate the explanation! @hendrikweisser

**Comment by zhuchcn** [comment]
When setting the `mappoing_tolerance` to 60, we found that most of the PSMs can be mapped to at least 1 feature. The mapping rate is over 95%. Below is the stdout. Our collaborator (who is a mass spec expert) is concerning that this is too high. Normally what would the normal mapping rate that you would see with FeatureFinderIdentification?

```
Summary statistics (counting distinct peptides including PTMs):
5046 peptides identified (5046 internal, 0 additional external)
4911 peptides with features (4911 internal, 0 external)
135 peptides without features (135 internal, 0 external)
```

**Comment by hendrikweisser** [comment]
I would agree that 60 s mapping tolerance may be too high. But it's not unreasonable to expect a high (> 90%) mapping rate, because the targeted feature detection should find features where the IDs (PSMs) are.
It may be worth checking some of the very basic settings ("extract:mz_window", "detect:peak_width") to see if the feature detection itself can be improved on your data. It can be very helpful to open your input mzML file, idXML file, and the output featureXML file on top of each other in TOPPView. Then you can really see where your features are relative to your PSMs, and where the raw data suggests that features should be detectable (or not). The optional "chrom_out" output file is also useful for troubleshooting, because you can directly see (again in TOPPView) the extracted ion chromatograms in which features (peaks) are detected. If you get very jagged curves (intensities dropping to zero), the "mz_window" parameter may be too low.

**Comment by hendrikweisser** [comment]
Looks like there were no further questions, so closing this issue.
